### PR TITLE
test: Adjust special cases for debian-stable moving to Debian 12 "bookworm"

### PR DESCRIPTION
**Pull request by martinpitt**
https://github.com/cockpit-project/bots/pull/4886 moves our "debian-stable" image from Debian 11 to 12. Adjust the test special cases.

Also enable the FreeIPA tests on Debian, as the packages are again available in both stable and testing. TestIPA.testQualifiedUsers still fails, so keep skipping that one for now -- let's unblock the lockstep bots+cockpit PR and investigate that as a follow-up.

---

This needs to land in lockstep with https://github.com/cockpit-project/bots/pull/4886

#18978 ran the libvirt change for all other OSes.

### CODE REVIEW

**Comment by martinpitt** [comment]
I'm starting with [testCheckpoint regression](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18975-20230620-073125-c5bb8e7a-debian-testing-bots-4886/log.html#48-2).

 * On the *old* image, eth0 appears in "Interfaces" (i.e. managed). It is configured via /etc/network/interfaces.d/50-cloud-init.  The image still has `ifupdown`, and `nmcli d` says "eth0    ethernet  connected (externally)".
 
 * On the *new* image, eth0 appears in "Unmanaged interfaces". it doesn't appear in `nmcli c`, and `nmcli d` says "eth0    ethernet  unmanaged". That image doesn't have `ifupdown`, nor /etc/network/interfaces*. Instead, the image has netplan, i.e. eth0 is managed by systemd-networkd. So it seems we should adjust the Debian images to move from netplan to NM as well now.

I cannot directly reproduce the `virsh net-list` failure that [breaks so many other tests](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18975-20230620-073125-c5bb8e7a-debian-testing-bots-4886/log.html), but it's plausibly related.

**Comment by martinpitt** [comment]
I can unreliably reproduce the [virsh net-list hang](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18975-20230620-111014-b5985eec-debian-testing-bots-4886/log.html) now. Running all the tests from the log works, but I left the VM sitting around for 20 mins, and now it hangs. Once it's in that situation, all `virsh` commands hang. Restarting libvirt helps.

File: test/verify/check-system-realms [hunk 1]
```diff
@@ -85,6 +85,7 @@ ExecStart=/bin/true
 @testlib.skipDistroPackage()
 class CommonTests:
     @testlib.timeout(900)
+    @testlib.skipImage("fails to leave the domain", "debian-*")
```
Comment (jelly) [inline test/verify/check-system-realms:88]: Do we have an issue for this?

**Review by jelly — APPROVED** [review]

File: test/verify/check-system-realms (hunk 1 above)
Comment (martinpitt) [reply test/verify/check-system-realms:88]: Not yet, so far just an entry on my TODO list, as it's not a regression.

**Pull request merged**